ci/jenkins/utils.sh

```diff
@@ -14,6 +14,17 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+function check_and_cleanup_buildkit_cache() {
+    free_space=$(df -h -B 1G / | awk 'NR==2 {print $4}')
+    free_space_threshold=40
+    if [[ $free_space -lt $free_space_threshold ]]; then
+        # If cleaning up dangling images unused in the last hour doesn't free up sufficient disk space,
+        # we will have to clean up all buildkit cache to release enough disk space.
```

Comment:
Do we really need to cleanup all cache? I think cache can still  help to speed up the image building. I saw there is an option `--keep-storage`, could you check if it's useful in this case? Thanks.

Reply:
@luolanzone  We can use this parameter to reduce the cache to 10GB during the first round cleanup. If the storage space is still insufficient, we will have to clean up all build cache.